deviceaccess.py

```diff
@@ -13,36 +13,81 @@
+                               pb.ScalarAccessor_uint8,
+                               pb.ScalarAccessor_uint16,
+                               pb.ScalarAccessor_uint32,
+                               pb.ScalarAccessor_uint64]
+oneDRegisterAccessor = Union[pb.OneDAccessor_float,
+                             pb.OneDAccessor_double,
+                             pb.OneDAccessor_int8,
+                             pb.OneDAccessor_int16,
+                             pb.OneDAccessor_int32,
+                             pb.OneDAccessor_int64,
+                             pb.OneDAccessor_uint8,
+                             pb.OneDAccessor_uint16,
+                             pb.OneDAccessor_uint32,
+                             pb.OneDAccessor_uint64]
+twoDRegisterAccessor = Union[pb.TwoDAccessor_float,
+                             pb.TwoDAccessor_double,
+                             pb.TwoDAccessor_int8,
+                             pb.TwoDAccessor_int16,
+                             pb.TwoDAccessor_int32,
+                             pb.TwoDAccessor_int64,
+                             pb.TwoDAccessor_uint8,
+                             pb.TwoDAccessor_uint16,
+                             pb.TwoDAccessor_uint32,
+                             pb.TwoDAccessor_uint64]
+registerAccessor = Union[scalarRegisterAccessor,
+                         oneDRegisterAccessor,
+                         twoDRegisterAccessor,
+                         pb.VoidRegisterAccessor]
+scalarRegister = NewType('scalarRegister', np.ndarray)
+oneDRegister = NewType('oneDRegister', np.ndarray)
+twoDRegister = NewType('twoDRegister', np.ndarray)
+voidRegister = NewType('voidRegister', np.ndarray)
+accessModeFlag = Optional[Sequence[AccessMode]]  # "Optional" adds "None" as legal type
+userTypes = Union[type, np.dtype]
+registerInputTypes = Union[np.ndarray, np.number, int, float]
+
+
+def setDMapFilePath(dmap_file_path: str) -> None:
+    """Set the location of the dmap file.
 
     The library will parse this dmap file for the device(alias) lookup.
     Relative or absolute path of the dmap file (directory and file name).
 
     Examples
     --------
     Setting the location of the dmap file
-      >>> da.setDMapFilePath("deviceInformation/exampleCrate.dmap")
+      >>> import deviceaccess as da
+      >>> da.setDMapFilePath('deviceInformation/exampleCrate.dmap')
+      >>> dmap_path = da.getDMapFilePath()
+      >>> print(dmap_path)
+      deviceInformation/exampleCrate.dmap
     """
-    pb.setDmapFile(dmapFilePath)
+    pb.setDmapFile(dmap_file_path)
 
 
 def getDMapFilePath() -> str:
-    """
-    Returns the dmap file name which the library currently uses for looking up device(alias) names.
-    """
+    """Returns the dmap file name which the library currently uses for looking up device(alias) names."""
     return pb.getDmapFile()
 
 
-class GeneralRegisterAccessor(ABC):
+class GeneralRegisterAccessor(ABC, np.ndarray):
```

Comment:
Please don't do that kind of cleanup now, it is conflicting with my pending changes and will be superseded by them!